### PR TITLE
Brown improve testing

### PR DESCRIPTION
**Pull request by bittirousku**
* Adds `conftest.py` which contains new testing fixtures that can be used everywhere in HEPCrawl tests.
* Improves Brown spider tests, adds validation and tests the final record after pipeline processing.
* Some PEP8 fixes.

### CODE REVIEW

File: tests/test_brown.py [hunk 1]
```diff
@@ -111,116 +201,99 @@ def test_abstract(record):
...
-    year = spider._get_phd_year(no_year_no_author)
+    assert record_no_year_no_author
+    assert "date_published" not in record_no_year_no_author
+    assert "authors" not in record_no_year_no_author
 
-    assert not year
 
-def test_no_author_in_thesis(no_year_no_author):
-    """Test that there are no authors."""
+def test_record_after_pipeline(record, patch_datetime_now):
```
Comment (bittirousku) [inline tests/test_brown.py]: So I'll just make this a fixture and return the `final_record` and use it everywhere in place or `record`?

File: tests/test_brown.py [hunk 2]
```diff
@@ -26,48 +27,137 @@
...
+@pytest.fixture
+def record(brown_splash_body, monkeypatch):
     """Return results from the Brown spider."""
     spider = brown_spider.BrownSpider()
-    response = fake_response_from_file('brown/test_1.json')
-    jsonresponse = json.loads(response.body_as_unicode())
-    jsonrecord = jsonresponse["items"]["docs"][0]
-    jsonrecord["uri"] = "brown/test_splash.html"
 
+    def patch_get_mime_type(monkeypatch):
```
Comment (bittirousku) [inline tests/test_brown.py]: Oh yeah, that's true.

File: tests/test_brown.py [hunk 3]
```diff
@@ -26,48 +27,155 @@
...
     response.meta["jsonrecord"] = jsonrecord
 
-    return spider.parse(response).next()
+    record = spider.parse(response).next()
+    pipeline = InspireAPIPushPipeline()
+    final_record = pipeline.process_item(record, spider)
 
-def test_files_constructed(parsed_node):
-    """Test pdf link.
+    return final_record
```
Comment (david-caro) [inline tests/test_brown.py]: Maybe we can refactor this to be reused on the tests, as it's repeated everywhere. Give it a couple hours, if you don't find a way to improve it, let's merge this and go for the arXiv one.

**Comment by jacquerie** [comment]
Stale, will be resurrected if/when needed.

**Pull request closed without merging**